### PR TITLE
Fuse batch normalization TensorFlow subgraphs in runtime

### PR DESCRIPTION
**Pull request by dkurt**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/445

### This pullrequest changes

resolves https://github.com/opencv/opencv/issues/10870
resolves https://github.com/opencv/opencv/issues/9894
resolves https://github.com/opencv/opencv/issues/11025

* Merge batch normalization and flatten TensorFlow subgraphs in runtime. Before model's import it fuses sets of nodes into a single alternative layer is known by importer.
* Batch normalization layer with no gamma parameter (no channel-wise multipliers)
* Squeeze operation for 4D blobs.

```
allow_multiple_commits=1
```

### CODE REVIEW

**Comment by vpisarev** [comment]
looks cool! But I would name "tf_graph_editor.*" somewhat differently. To me "graph editor" usually means an interactive application (or component) with UI that lets user edit the graph.

what about "tf_graph_simplify.*"?

**Comment by dkurt** [comment]
@vpisarev, Thanks! Replaced to `tf_graph_simplifier`.
